public/app/features/dashboard-scene/serialization/transformSaveModelSchemaV2ToScene.test.ts

```diff
@@ -85,14 +112,72 @@ describe('transformSaveModelSchemaV2ToScene', () => {
     expect(dashboardControls.state.refreshPicker.state.intervals).toEqual(time.autoRefreshIntervals);
     expect(dashboardControls.state.hideTimeControls).toBe(time.hideTimepicker);
 
-    // TODO: Variables
-    // expect(scene.state?.$variables?.state.variables).toHaveLength(dash.variables.length);
-    // expect(scene.state?.$variables?.getByName(dash.variables[0].spec.name)).toBeInstanceOf(QueryVariable);
-    // expect(scene.state?.$variables?.getByName(dash.variables[1].spec.name)).toBeInstanceOf(TextBoxVariable); ...
+    // Variables
+    const variables = scene.state?.$variables;
+    expect(variables?.state.variables).toHaveLength(dash.variables.length);
+    validateQueryVariable({
+      variable: variables?.state.variables[0],
+      kind: dash.variables[0] as QueryVariableKind,
```

Comment:
`kind` is confusing here, let's name it `variableSpec`

Reply:
I've renamed this to `variableKind` because we are passing kinds like `QueryVariableKind` from the schema, and I also need it to in `validateVariable` function to do stuff like this:  
```ts
if (sceneVariable instanceof DataSourceVariable && variableKind.kind === 'DatasourceVariable') {
    expect(sceneVariable?.state.pluginId).toBe(variableKind.spec.pluginId);
  }
```